common/protob/messages-ethereum.proto

```diff
@@ -111,6 +111,46 @@ message EthereumMessageSignature {
     required string address = 3;     // address used to sign the message
 }
 
+
+/**
+ * Request: Ask device to sign typed data
+ * @start
+ * @next EthereumTypedDataRequest
+ * @next Failure
+ */
+message EthereumSignTypedData {
+    repeated uint32 address_n = 1;           // BIP-32 path to derive the key from master node
+    optional bool use_v4 = 2 [default=true]; // use EIP-712 (v4) for typed data signing
```

Comment:
Metamask has its own implementation history, all v1-v2-v3 were not compliant with the latest standard and should not be considered.

Their implementation `v4` is widely adopted now, but still, has a bug described there:
https://github.com/MetaMask/eth-sig-util/pull/107#issuecomment-765620780

So at one day, it might be a `v5`. 

I agree that having version enumeration is better than trying to maintain only a single or two "true" versions. I'm not a big fan of maintaining that as part of HW, but we have no choice there.